source/common/router/config_impl.cc

```diff
@@ -467,6 +467,22 @@ VirtualHostImpl::VirtualClusterEntry::VirtualClusterEntry(const Json::Object& vi
   priority_ = ConfigUtility::parsePriority(virtual_cluster);
 }
 
+const VirtualHostImpl* RouteMatcher::findWildcardVirtualHost(const std::string& host) const {
+  for (const auto& iter : wildcard_virtual_host_suffixes_) {
```

Comment:
it's not immediately intuitive to me the algorithm being used here (have to stare at it a bunch). If possible can you add a slightly larger overview comment saying what this function is about to do.